server/src/controllers/ImagesController.ts

```diff
@@ -1,34 +1,65 @@
 import { getRepository } from 'typeorm';
 import { Image } from '../database/entities/Image';
+import { ImageType } from '../database/entities/ImageType';
 import {
   Response, Controller, Get, Post, Request, Delete, Params
 } from '@decorators/express';
 import {
   Response as ExpressResponse,
   Request as ExpressRequest
 } from 'express';
+import { ImageUploadMiddleware } from '../http/ImageUploadMiddleware';
 
 @Controller('/images')
 export class ImagesController {
   @Get('/')
   async list(
     @Response() res: ExpressResponse
   ) {
-    const records = await getRepository(Image).find();
+    const records = await getRepository(Image).find({
+      order: { id: "DESC" },
+      relations: [ 'imageType' ],
+    });
 
-    res.send({ images: records });
+    res.send({
+      images: records.map(({
+        id,
+        imageType,
+        fileName,
+        filePath,
+        src,
+      }) => ({
+        extension: imageType.extension,
+        fileName,
+        filePath,
+        id,
+        type: imageType.mimeType,
+        src,
+      })).sort((a, b) => b.id - a.id)
+    });
   }
-  @Post('/')
+  @Post('/', [ ImageUploadMiddleware ])
   async create(
-    @Request() _req: ExpressRequest,
+    @Request() req: ExpressRequest,
     @Response() res: ExpressResponse
   ) {
     const image = new Image();
+    const imageType = await getRepository(ImageType).findOne({
+      mimeType: req.file.mimetype
+    })
+    let success = false;
 
-    image.filePath = `/new-image-path/image-${Date.now()}.png`;
-    await getRepository(Image).save(image);
+    if (req.file && imageType) {
+      image.filePath = `${req.file.path}`;
+      image.imageType = imageType;
+      await getRepository(Image).save(image);
+      success = true;
```

Comment:
I'd rather throw status code (400) when something is wrong, this is not really REST standard.

Reply:
we can of course throw a custom RequestError, which will be picked up by the ErrorHandler... the default fail status is 500, do you want to change that to 400?

Reply:
Sure, has to be 400 so we can render correct error on Fe.